pkg/router/controller/host_admitter.go

```diff
@@ -227,8 +231,18 @@ func (p *HostAdmitter) displacedRoutes(newRoute *routeapi.Route) ([]*routeapi.Ro
 	// See if any existing wildcard routes block our domain, or if we displace them
 	for i, route := range p.claimedWildcards[wildcardKey] {
 		if route.Namespace == newRoute.Namespace {
-			// Never displace a route in our namespace
-			continue
+			// If new route is a wildcard, still need to check
+			// if it is older as it can displace newer wildcard
+			// routes in the same namespace.
+			if newRoute.Spec.WildcardPolicy != routeapi.WildcardPolicySubdomain || route.Name == newRoute.Name {
```

Comment:
Adding a few duplicate host+path and wildcard+path routes into the fuzz tests would be good as well